Different branch for notebook-latex report integration?

a-t-0 opened the pull request:
Dear Dr.ir. D. Dirkx,

This is a template for the 3 assignments of the Numerical Astrodynamics course of 2020. It automatically runs the jupyter notebooks, converts them to pdf.

I am not yet sure whether you also expect a proper report along with the notebooks, but they are also integrated into a latex report that is also automatically compiled.

Perhaps some students would find this convenient, so it might be nice to put this template into a separate branch of your repository for those that want it. If so, please decline this pull request, and create a new branch, then I will send a pull request towards that branch. 

Note:
0. It can still be improved by writing a unit test that tests a function inside the jupyter notebook
1, The continuous integration testing with Travis is not yet enabled because I did not yet set the `.travis.yml` up correctly. 
2, This pull request does not contain any solutions to the assignments.
3. I did not test it on a clean installation of Ubuntu, so If there is a package that is not automatically installed (E.g. if my device already had it without me remembering I had the module/package) please let me know, so I can add the dependencies to the respective `..._environment.yml` files.

Commit:
Set up CI.

Commit:
Created basic report structure.

Commit:
Removed original notebook file duplicates.

Commit:
Renamed hw folders to project folders.

Commit:
Updated gitignore.

Commit:
Made first assignment latex compilable.

Commit:
Updated readme.md.

a-t-0 committed:
Update README.md

a-t-0 committed:
Improved layout.

a-t-0 committed:
Cleaned readme.md

a-t-0 committed:
Included CI badge.

Commit:
Included yml for travis-CI.

a-t-0 committed:
Update README.md

a-t-0 committed:
Removed typo.

Commit:
Made jupyter notebooks runable from main.

Commit:
Created automatic converting jupyter notebook to pdf.

Commit:
Enabeled auto compilation of latex project1.

Commit:
Tested latex compilation and pdf moving.

Commit:
Included jupyter notebook into pdf.

Commit:
Refactored code.

Commit:
Verified compilation for all 3 assignments.

Commit:
Included dependencies of original yaml into travis yaml.

Commit:
Built environment from other yml in travis yaml.

Commit:
Included getting conda in travis CI.

Commit:
Removed typo.

Commit:
Removed sklearn pip module.

a-t-0 committed:
Updated readme.

a-t-0 committed:
Update README.md

a-t-0 committed:
Update README.md

a-t-0 committed:
Update README.md

Commit:
Updated travis yml.

Commit:
Included a unit test inside the jupyter notebook.

Commit:
Deleted old files.

Commit:
Reverted to previous commit to prevent erroneous file deletion.

a-t-0 committed:
Changed pytest syntax.

Commit:
Included matplotlib module.

DominicDirkx commented:
Hi,

This looks like a great piece of work! We're trying out the use of Jupyter notebooks for the first time this year, and will likely be using several different formats for the assignments to see what works best, and which format is most liked by students and by us. To clarify: for assignment 1, you are to submit 7 different notebooks, and 1 pdf report. For assignment 2, we will likely take a different route, and ask for a single notebook which incorporates the assignment given to you, the code used to generate the results, and your answers to the questions.

Our time to work on this is sadly very limited. Our first and foremost concern is to make sure everything is working in our new code setup, and in the online setup, and I'm not sure if we'll have the time to properly review your work to incorporate it into this year's run of the course. That being said, if the code you've written is properly documented, and others can understand how to use it, it would be great if it is of use to them in addition to yourself. But, until we've had a chance to properly go through it ourselves, we will not be able to actively provide support on its use. Once my health and agenda start clearing up a bit, hopefully in the coming weeks, I'd love to discuss your ideas on this further, so we can most effectively use you work going forward,

Best,

Dominic

a-t-0 committed:
Included nb_conda_kernels in yml for travis CI.

a-t-0 committed:
Included ipykernel module to yaml for travis-CI.

a-t-0 committed:
Included tudatpy as duplicate for travis-CI.

a-t-0 committed:
Test whether module can be ommitted.

a-t-0 committed:
Test whether tudatpy dependency can be ommited.

a-t-0 committed:
Renamed environment file.

a-t-0 committed:
Delete linux_environment.yml

Currently there is no difference between Windows and Linux setup. MAC setup is not tested.

a-t-0 committed:
Updated name of environment.yml.

a-t-0 committed:
Ommitted superfluous modules in yml.

a-t-0 committed:
Simplified readme instructions.

Commit:
Wrote more documentation.

Commit:
Included pyment generated docstrings.

Commit:
Corrected relative output path of jupyter notebook images.

Commit:
Restored unit testing functionality.

Commit:
Included documentation generation tools.

Commit:
Generated html documentation with pdoc.

Commit:
Verfied all three assignment templates.

Commit:
Included auto-generated html documentation for assignments 2 and 3.

Commit:
Propagated latest change to notebook from original repo.

a-t-0 commented:
Dear Dr.ir. D. Dirkx,

Thank you for your clear and complete response. I hope your health recovers quickly and completely. This is a notification to provide an updated functionality description of the fork of your repository. I did *not* upstream those updates into this pull request. I understand you might not have time to verify the code, develop and provide support for this setup this year.

In the fork, I have rewritten and included documentation in the docstrings of the `.py` files based on the documentation templates generated by `pyment`. Html documentation is generated using `pdoc` for all but the `Main.py` file, as the generation of the html documentation of `Main.py` raised an error.

The unit tests and accompanying  continuous integration in Travis-CI verify the main codes runs the jupyter notebooks, and compiles the latex report templates.

If you have any questions, feel free to raise an issue, or respond to this thread.

a-t-0 committed:
Updates from Overleaf

a-t-0 committed:
Merge overleaf-2020-12-13-0314 into master

Commit:
Included hw2 assignment.

Commit:
Merge branch 'master' of https://github.com/a-t-0/NumericalAstrodynamicsAssignments_2020
